### DISCUSSION TITLE
Pattern related question -  big state size

### DISCUSSION THREAD

**Discussion by imbolc** [davidpdrsn/axum-live-view#142] [General]
I'm wondering about a way to implement something like infinite scroll. I've look into the chat example and messages there are shared between all users. But in case of infinite scroll it's state could be different for every user, so each handler would have its own list of potentially infinite growing items and memory consumptions. Is there a way to avoid this?

**Comment by davidpdrsn** [comment]
The chat example is intentionally quite naive. In a real world scenario you wouldn't keep all the message in memory but in a database somewhere. Then its just a matter of querying the database and loading messages into the view based on scroll events. You'd probably like remove messages from memory that are far away from the current position, like you're hinting at.

I've not actually implemented something like this but does that somewhat answer your question?

**Comment by imbolc** [comment]
Let's look at youtube for example. You can adjust some filters there, and then when you scroll the page down, you'll see more and more videos appended to the page without removing previous.

We can keep previously loaded items in the memory of the handler and load only new items from db. It would mean unpredictable memory consumption.

Or we can fetch all the items (including previous once) from db each time, it would lead to high db load.

Also in both cases we would render unpredictable huge page each time, which would impact cpu load.

It just feels like a significant waste of resources in contrast to the classic approach, when you load each portion of data from db separately and send it to the client to add to the page. So I wonder if there's a way to avoid it.

**Comment by davidpdrsn** [comment]
🤷

**Comment by davidpdrsn** [comment]
I'm not sure. I haven't built something like that so really can't say.

**Comment by davidpdrsn** [comment]
You could try to google for how infinite scrolling is implemented in Phoenix LiveView, which axum-live-view is inspired by. Probably lots of good resources about it which should translate pretty easily.
